src/Outsystems.SetupTools/Functions/Publish-OSPlatformSolutionPack.ps1

```diff
@@ -0,0 +1,178 @@
+    Publish-OSPlatformSolution -ServiceCenterHost "8.8.8.8" -Solution 'c:\solution.osp' -Credential $Credential -Wait
+
+    .EXAMPLE
+    $Credential = Get-Credential
+    Publish-OSPlatformSolution -ServiceCenterHost "8.8.8.8" -Solution 'c:\solution.osp' -Credential $Credential -StopOnWarnings
+
+    .NOTES
+    You can run this cmdlet on any machine with HTTP access to Service Center.
+
+    The cmdlet will return an object with an ExitCode property that will match one of the following values:
+    -1 = Error while trying to publish the solution
+    0  = Success
+    1  = Solution published with warnings
+    2  = Solution published with errors
+
+    This cmdlet does not check the integrity of the solution pack before starting.
+    Trusts on the Service Center to make all the checks.
+
+    #>
+
+    [CmdletBinding(DefaultParameterSetName = 'PSCred')]
+    [OutputType('Outsystems.SetupTools.PublishResult')]
+    param (
+
+        [ValidateNotNullOrEmpty()]
+        [string]$Solution,
+
+        [Parameter(ParameterSetName = 'UserAndPass')]
+        [ValidateNotNullOrEmpty()]
+        [string]$ServiceCenterUser = $OSSCUser,
+
+        [Parameter(ParameterSetName = 'UserAndPass')]
+        [ValidateNotNullOrEmpty()]
+        [string]$ServiceCenterPass = $OSSCPass,
+
+        [Parameter(ParameterSetName = 'PSCred')]
+        [ValidateNotNullOrEmpty()]
+        [System.Management.Automation.Credential()]
+        [System.Management.Automation.PSCredential]$Credential = $OSSCCred
+    )
+
+    begin
+    {
+        LogMessage -Function $($MyInvocation.Mycommand) -Phase 0 -Stream 0 -Message "Starting"
+        SendFunctionStartEvent -InvocationInfo $MyInvocation
+
+        # Initialize the results object
+        $publishResult = [pscustomobject]@{
+            PSTypeName = 'Outsystems.SetupTools.PublishResult'
+            PublishId  = 0
+            Errors     = 0
+            Warnings   = 0
+            Success    = $true
+            ExitCode   = 0
+            Message    = 'Solution successfully installed'
+        }
+        $osVersion = GetServerVersion
+        $osInstallDir = GetServerInstallDir
+
+        switch ($PsCmdlet.ParameterSetName)
+        {
+            "PSCred"
+            {
+                $ServiceCenterUser = $Credential.UserName
+                $ServiceCenterPass = $Credential.GetNetworkCredential().Password
+            }
+        }
+    }
+
+    process
+    {
+
+        if ($(-not $osVersion) -or $(-not $osInstallDir))
+        {
+            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 3 -Message "Outsystems platform is not installed"
+            WriteNonTerminalError -Message "Outsystems platform is not installed"
+
+            $publishResult.Success = $false
+            $publishResult.ExitCode = -1
+            $publishResult.Message = 'Outsystems platform is not installed'
+
+            return $publishResult
+        }
+
+        if ($(GetSCCompiledVersion) -ne $osVersion)
```

Comment:
This is wrong. This is checking the service center installed version. This function is only used in the Publish-OSPlatformServiceCenter